activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -78,7 +78,7 @@ def initialize(constructor = {})
     def default(*args)
       arg_key = args.first
 
-      if include?(key = convert_key(arg_key))
+      if include?(key = convert_key(arg_key)) && args.size > 0
         self[key]
```

Comment:
`default` vs `default(nil)` behaved poorly even before the less-than-ideal `dig` fix.  My inclination is to leave that portion as is, and once this is merged, I'd be more than happy to pr a better `dig` with conforming `default`.